File: easy/python/longest_common_prefix.py
from ast import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        prefix = strs[0]
        for i in range(1, len(strs)):
            temp = ""
            for j in range(min(len(prefix), len(strs[i]))):
                if prefix[j] == strs[i][j]:
                    temp += prefix[j]
                else:
                    break
            prefix = temp
        return prefix

File: easy/ruby/longest_common_prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
  return '' if strs.empty?

  prefix = strs[0]
  strs.each do |str|
    prefix = prefix.chop until str.start_with?(prefix)
  end

  prefix
end
